### PR TITLE
Add BigLoop benchmark.

### PR DESCRIPTION
**Pull request by vext01**
This will serve as our first optimisation target.

We are still 3x slower than Lua, even with -O2.

(I copied the licensing from the main yk repo, which appears to be dual apache/mit)

(It's almost not worth licensing at this stage)

### CODE REVIEW

File: rebench.conf [hunk 1]
```diff
@@ -40,6 +40,22 @@ benchmark_suites:
...
             - Towers:
                 extra_args: 600
+    # Benchmarks written "in-house".
+    yk:
+        gauge_adapter: RebenchLog
+        # Borrowing the harness from awfy.
+        command: " ../../awfy/Lua/harness.lua %(benchmark)s %(iterations)s "
+        max_invocation_time: 600 # seconds per vm invocation
+        min_iteration_time: 200 # miliseconds per iteration
+        invocations: 1 # the number of process executions
```
Comment (vext01) [inline rebench.conf:50]: gah, this is supposed to be higher -- bear with

Comment (vext01) [reply rebench.conf]: fixed in 1b02e8e

**Comment by vext01** [comment]
(will need squash)

**Comment by ptersilie** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Comment by vext01** [comment]
>  fatal: destination path 'yk' already exists and is not an empty directory

hah! that's quite amusing. I called the new suite `yk` and we then can't clone the`yk` repo.

**Comment by ptersilie** [comment]
Please squash,

**Comment by vext01** [comment]
squashed.

**Comment by vext01** [comment]
This can go in now.

**Comment by vext01** [comment]
```
14:06:51 + cargo run results html
14:06:51     Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
14:06:51      Running `target/debug/reporter results html`
14:06:51 thread 'main' panicked at src/main.rs:75:25:
```

something to do with the reporter, investigating...

**Comment by vext01** [comment]
OK to squash?

**Comment by vext01** [comment]
squashed

**Pull request merged**